File: client/src/types/JsonModels/Subscriptions/CurrentSubscriptionModel.ts
import type { JsonModels } from "../../backend";
import type { EntitlementSource } from "../Enums/EntitlementSource";
import type { SubscriptionStatus } from "../Enums/SubscriptionStatus";

export interface CurrentSubscriptionModel
{
	planId: number;
	planCode: string;
	planName: string;
	source: EntitlementSource;
	status?: SubscriptionStatus;
	currentPeriodEnd?: string;
	cancelAtPeriodEnd: boolean;
	features: JsonModels.Subscriptions.FeatureAvailabilityModel[];
}

File: server/FitMate.Core/JsonModels/Subscriptions/CurrentSubscriptionModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.Subscriptions;

public class CurrentSubscriptionModel
{
    public long PlanId { get; set; }
    public string PlanCode { get; set; } = string.Empty;
    public string PlanName { get; set; } = string.Empty;
    public EntitlementSource Source { get; set; }
    public SubscriptionStatus? Status { get; set; }
    public DateTime? CurrentPeriodEnd { get; set; }
    public bool CancelAtPeriodEnd { get; set; }
    public List<FeatureAvailabilityModel> Features { get; set; } = [];
}
